arangod/VocBase/vocbase.cpp

```diff
@@ -1764,14 +1763,27 @@ std::string TRI_vocbase_t::path() const {
   return engine->databasePath(this);
 }
 
+bool TRI_vocbase_t::IsAllowedName(arangodb::velocypack::Slice slice) noexcept {
+  return !slice.isObject()
+    ? false
+    : IsAllowedName(
+        arangodb::basics::VelocyPackHelper::readBooleanValue(slice, "isSystem", false),
+        arangodb::basics::VelocyPackHelper::getStringRef(slice, "name", "")
+      )
+    ;
+}
+
 /// @brief checks if a database name is allowed
 /// returns true if the name is allowed and false otherwise
-bool TRI_vocbase_t::IsAllowedName(bool allowSystem, std::string const& name) {
+bool TRI_vocbase_t::IsAllowedName(
+    bool allowSystem,
+    arangodb::velocypack::StringRef const& name
+) noexcept {
```

Comment:
are you sure this can't throw ?

Reply:
ah nevermind

Reply:
I don't see anything in the code that can throw, have I missed something?